server/src/routes/groupExpenses.ts

```diff
@@ -0,0 +1,213 @@
+import express, { Request, Response } from 'express';
+import prisma from '../prismaClient';
+import { body, validationResult } from 'express-validator';
+import { authenticate } from '../middleware/authenticate';
+import dotenv from 'dotenv';
+import { ExpenseCategory } from '@prisma/client';
+
+dotenv.config();
+
+const router = express.Router();
+
+/**
+ * Create a new group expense
+ *
+ * This route creates a new group expense for the authenticated user.
+ *
+ * @route POST /group-expenses
+ * @param title Title of the group expense
+ * @param amount Amount of the group expense
+ * @param category Category of the group expense
+ * @param createdAt Date the group expense was created
+ * @param usernames Usernames of the users that are part of the group expense
```

Comment:
I think this should be changed to splits and define what a split is(?)
```suggestion
 * @param split splits are (key, value) pairs with the usernames of the users that are part of the group expense as keys and value is the percentage as a decimal of what they need to pay
```

Reply:
TY good catch, fixed in new commit